### DISCUSSION TITLE
[Question] Long running tasks and app restart for update

### DISCUSSION THREAD

**Discussion by jfsimon** [OptimalBits/bull#2563] [General]
Hello,

First of all, thank you for this library. I'm in love with Redis and Bull is a splendid queue manager built on it :)
I know my question could seem a little naïve, but I cant find any resource on this topic.

I have long running tasks (up to several hours), it's working great but I wonder how to deal with app updates.
When updating application, I need to restart the main consumer process, and it kills the tasks processes.
Has anyone a trick to handle theses updates gracefully without restarting the jobs?
Maybe using `PM2` or something like that... If someone has already solved this problem, I'm all ear!

**Comment by manast** [comment]
So if you must restart the main process without waiting for the jobs to complete then they will get stalled and picked up by another worker... there is no solution to your problem other than waiting for the current jobs to complete, this is handled by ```close``` https://docs.bullmq.io/guide/going-to-production#gracefully-shut-down-workers

**Comment by jfsimon** [comment]
Thank you @manast for converting this into discussion, I didn't noticed they were enabled.
I don't know about the internals of Bull, maybe I should dig a little.

Don't you think there is a way to do something like that:
- listen for new jobs (the main thread would not be a consumer)
- start a new process using `PM2` (which is a consumer),
- handle the job through this new process,
- kill the process when done?

Thanks for the help!
